x/gov/keeper/tally.go

```diff
@@ -112,15 +104,55 @@ func (keeper Keeper) Tally(ctx context.Context, proposal v1.Proposal) (passes, b
 		totalVotingPower = totalVotingPower.Add(votingPower)
 	}
 
-	params, err := keeper.Params.Get(ctx)
+	return totalVotingPower, results, nil
+}
+
+// getCurrentValidators fetches all the bonded validators, insert them into currValidators
+func (k Keeper) getCurrentValidators(ctx context.Context) (map[string]v1.ValidatorGovInfo, error) {
+	currValidators := make(map[string]v1.ValidatorGovInfo)
+	if err := k.sk.IterateBondedValidatorsByPower(ctx, func(index int64, validator stakingtypes.ValidatorI) (stop bool) {
+		valBz, err := k.sk.ValidatorAddressCodec().StringToBytes(validator.GetOperator())
+		if err != nil {
+			return false
+		}
+		currValidators[validator.GetOperator()] = v1.NewValidatorGovInfo(
+			valBz,
+			validator.GetBondedTokens(),
+			validator.GetDelegatorShares(),
+			math.LegacyZeroDec(),
+			v1.WeightedVoteOptions{},
+		)
+
+		return false
+	}); err != nil {
+		return nil, err
+	}
+
+	return currValidators, nil
+}
+
+// Tally iterates over the votes and updates the tally of a proposal based on the voting power of the
+// voters
+func (k Keeper) Tally(ctx context.Context, proposal v1.Proposal) (passes, burnDeposits bool, tallyResults v1.TallyResult, err error) {
+	currValidators, err := k.getCurrentValidators(ctx)
 	if err != nil {
-		return false, false, tallyResults, err
+		return false, false, tallyResults, fmt.Errorf("error while getting current validators: %w", err)
+	}
+
+	tallyFn := k.calculateVoteResultsAndVotingPowerFn
+	if tallyFn == nil {
+		tallyFn = defaultCalculateVoteResultsAndVotingPower
 	}
```

Comment:
can simplify a bit since we can expect it to be set now
```suggestion
	tallyFn := k.calculateVoteResultsAndVotingPowerFn
```